### ISSUE TITLE
[FEA] Adjust libcudf to use kvikIO for small host reads

### ISSUE THREAD

**Issue by GregoryKimball** [rapidsai/cudf#17259]
**Is your feature request related to a problem? Please describe.**

For parquet columns that are highly compressible, the page size can be very small. For example, we observe 50-100 KB pages for the NDS-H column `l_shipinstruct` in `lineitems` with ZSTD compression. This column has a cardinality of 4, encodes well with run_length, and (may) show other compression opportunities as well. 

```
>>> df['l_shipinstruct'].unique()
0    DELIVER IN PERSON
1     TAKE BACK RETURN
2                 NONE
3          COLLECT COD
```

When performing IO on these pages, libcudf falls back to a host read from pageable instead of a pinned read via kvikIO. 

For an async memory resource, using host read from pageable is not that different than the kvikIO option. However, when using a managed memory pool resource, host read from pageable appears to be ~2x slower in the IO step.

![Image](https://github.com/user-attachments/assets/46ab0761-2ed5-4e44-ba98-a745d6360e5f)

**Describe the solution you'd like**
We should consider adjusting libcudf to always use kvikIO instead.

I believe we should consider refactoring datasource to not use a read threshold, and also decouple the `gds` labels from dispatch to kvikIO for host reads.
```
static constexpr size_t _gds_read_preferred_threshold = 128 << 10;  // 128KB
```

**Additional context**

We've collected evidence that with a managed memory pool, the HtoD host copy from `pageable to managed` is slower than the copy from `pageable to device`, at least for these small pages.

**Comment by GregoryKimball** [comment]
@brandon-b-miller @vuule thank you for the discussion today about the pageable copy difference with a managed pool MR.

**Comment by vuule** [comment]
Related issue https://github.com/rapidsai/cudf/issues/17228: use kvikIO for host reads.
Most likely not required to avoid the pageable copies in this case.

**Comment by vuule** [comment]
Opened https://github.com/rapidsai/cudf/pull/17260 with a potential fix.
Haven't evaluated the performance impact.

**Comment by vyasr** [comment]
Resolved by #17260 (@vuule @GregoryKimball please reopen if needed).
